README.md

```diff
@@ -0,0 +1,155 @@
+# Vault integration for Python apps
+
+Applications need to be able to obtain sensitive information from Vault
+via its API. Handling this outside of the application is a bit janky; either a
+sidecar process is required, or the app must be periodically killed and
+restarted with fresh secrets (e.g. database credentials).
+
+The goal of this project is to reduce the complexity of using vault secrets
+implementing a lightweight integration. It should accept the following
+parameters:
+
+- Auth token (can be passed in as an env var)
+- Vault endpoint and port number
+- The name of the Vault role that the application requires
+
+It should be configurable to retrieve the followings kinds of secrets:
+- AWS credentials bound to a specific policy in Vault
+- Database credentials provisioned by Vault
+- Generic secrets (key/value)
+
+This package should provide a secrets manager that retrieves secrets, holds on
+to them, monitors the passage of time relative to the lease duration on each
+secret that it retrieves, and automatically refresh the credential(s) as
+needed.
+
+To use this in WSGI applications, we should also provide a WSGI middleware.
```

Comment:
"should provide" or "also provide"? making sure docs reflect reality.

Reply:
Rephrased

> So that we can easily use this package in WSGI applications, it should also
provide a WSGI middleware.